## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#faf5f5c5-2151-4111-a8ea-f976ee5e96a0) |
| mint-fs.sh | :x: [more...](#b6d4e594-8c37-41ca-a4c1-724e169ef4e9) |
| mint-erasure.sh | :x: [more...](#9dd71fa2-d02e-42cf-a0b2-e6549b45579b) |
| mint-dist-erasure.sh | :x: [more...](#65626144-dad5-4514-b754-de9c8d4ae3fe) |
| mint-gateway-nas.sh | :x: [more...](#2ebdd052-b893-4dea-9d12-a051cdb8209f) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#b2c72418-cefc-456e-8ffe-6191f4a11df5) |
<a name="2ebdd052-b893-4dea-9d12-a051cdb8209f"></a>
### 12009-8b2eb4e/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30853
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1dacb9ff1639:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 40 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 1 minutes and 7 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 38 seconds
(8/15) Running minio-dotnet tests ... done in 28 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 24 seconds
(10/15) Running minio-java tests ... done in 12 seconds
(11/15) Running minio-js tests ... done in 41 seconds
(12/15) Running minio-py tests ... done in 1 minutes and 2 seconds
(13/15) Running s3cmd tests ... done in 17 seconds
(14/15) Running s3select tests ... FAILED in 5 minutes and 2 seconds
{
  "name": "s3select:test_sql_operators",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "total_tests": 20,
    "total_success": 15
  },
  "duration": 300371,
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-e7662bd2-80a8-4134-87c3-d4e647bb6a12, request_id: 1673A4708790B448, host_id: 90b2e972-31fe-4b22-ab40-52a7b47e08e4, bucket_name: s3select-test-e7662bd2-80a8-4134-87c3-d4e647bb6a12",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 61, in main\n    test_sql_operators(client, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 140, in test_sql_operators\n    raise select_err\n  File \"/mint/run/core/s3select/sql_ops.py\", line 138, in test_sql_operators\n    test_sql_expressions(client, json_testfile, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 91, in test_sql_expressions\n    input_serialization, output_serialization, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 78, in test_sql_expressions_custom_input_output\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-e7662bd2-80a8-4134-87c3-d4e647bb6a12, request_id: 1673A4708790B448, host_id: 90b2e972-31fe-4b22-ab40-52a7b47e08e4, bucket_name: s3select-test-e7662bd2-80a8-4134-87c3-d4e647bb6a12\n",
  "status": "FAIL"
}
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="faf5f5c5-2151-4111-a8ea-f976ee5e96a0"></a>
### 12009-8b2eb4e/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30800
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ce7411d22f4b:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 1 minutes and 26 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 30 seconds
(8/15) Running minio-dotnet tests ... done in 47 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 16 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 13 seconds
(11/15) Running minio-js tests ... done in 45 seconds
(12/15) Running minio-py tests ... done in 2 minutes and 30 seconds
(13/15) Running s3cmd tests ... done in 17 seconds
(14/15) Running s3select tests ... FAILED in 1 minutes and 3 seconds
{
  "name": "s3select:test_sql_operators",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "total_tests": 20,
    "total_success": 15
  },
  "duration": 61123,
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-469d70b8-c66a-4b31-bedb-3db4f812596d, request_id: 1673A470DC75E162, host_id: b8c16bd4-a01e-4955-84dc-eb864c068ff9, bucket_name: s3select-test-469d70b8-c66a-4b31-bedb-3db4f812596d",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 61, in main\n    test_sql_operators(client, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 140, in test_sql_operators\n    raise select_err\n  File \"/mint/run/core/s3select/sql_ops.py\", line 138, in test_sql_operators\n    test_sql_expressions(client, json_testfile, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 91, in test_sql_expressions\n    input_serialization, output_serialization, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 78, in test_sql_expressions_custom_input_output\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-469d70b8-c66a-4b31-bedb-3db4f812596d, request_id: 1673A470DC75E162, host_id: b8c16bd4-a01e-4955-84dc-eb864c068ff9, bucket_name: s3select-test-469d70b8-c66a-4b31-bedb-3db4f812596d\n",
  "status": "FAIL"
}
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="b2c72418-cefc-456e-8ffe-6191f4a11df5"></a>
### 12009-8b2eb4e/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:30737
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 602099746e71:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2870,
  "function": "aws --endpoint-url http://minio-c3.minio.io:30737 s3api copy-object --bucket awscli-mint-test-bucket-6141 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-6141/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 47 seconds
(7/15) Running minio-dotnet tests ... done in 39 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 7 seconds
{
  "args": {},
  "duration": 283,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 38 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 349,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-1nrf6p1, objectName = minio-java-test-3bp0quo-copy, resource = /minio-java-test-1nrf6p1/minio-java-test-3bp0quo-copy, requestId = 1673A429E702CA63, hostId = 61900965-6643-4479-8943-7bdcfff7e8e8)\nrequest={method=PUT, url=http://minio-c3.minio.io:30737/minio-java-test-1nrf6p1/minio-java-test-3bp0quo-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-2cirdn/minio-java-test-3bp0quo\nHost: minio-c3.minio.io:30737\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210407T173951Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210407/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Wed, 07 Apr 2021 17:39:50 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1673A429E702CA63\nX-Xss-Protection: 1; mode=block\nDate: Wed, 07 Apr 2021 17:39:51 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 50 seconds
(9/15) Running minio-py tests ... done in 3 minutes and 10 seconds
(10/15) Running s3cmd tests ... FAILED in 5 seconds
{
  "name": "s3cmd",
  "duration": "2846",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-27106/s3cmd-test-object-27714 5682b638-a5ad-44a4-a8a8-c4da5bf28545\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 5682b638-a5ad-44a4-a8a8-c4da5bf28545 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... FAILED in 1 minutes and 4 seconds
{
  "name": "s3select:test_sql_operators",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "total_tests": 20,
    "total_success": 15
  },
  "duration": 61280,
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-940c2464-dd92-4c2f-aba0-88a22ea1715b, request_id: 1673A471E362FBEF, host_id: 61900965-6643-4479-8943-7bdcfff7e8e8, bucket_name: s3select-test-940c2464-dd92-4c2f-aba0-88a22ea1715b",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 61, in main\n    test_sql_operators(client, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 140, in test_sql_operators\n    raise select_err\n  File \"/mint/run/core/s3select/sql_ops.py\", line 138, in test_sql_operators\n    test_sql_expressions(client, json_testfile, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 91, in test_sql_expressions\n    input_serialization, output_serialization, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 78, in test_sql_expressions_custom_input_output\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-940c2464-dd92-4c2f-aba0-88a22ea1715b, request_id: 1673A471E362FBEF, host_id: 61900965-6643-4479-8943-7bdcfff7e8e8, bucket_name: s3select-test-940c2464-dd92-4c2f-aba0-88a22ea1715b\n",
  "status": "FAIL"
}
(10/15) Running security tests ... done in 0 seconds

Executed 10 out of 15 tests successfully.
```
<a name="65626144-dad5-4514-b754-de9c8d4ae3fe"></a>
### 12009-8b2eb4e/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30168
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4aa7c4d20b70:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 2 minutes and 15 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 8 seconds
(8/15) Running minio-dotnet tests ... done in 46 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 39 seconds
(10/15) Running minio-java tests ... done in 54 seconds
(11/15) Running minio-js tests ... done in 46 seconds
(12/15) Running minio-py tests ... done in 2 minutes and 29 seconds
(13/15) Running s3cmd tests ... done in 16 seconds
(14/15) Running s3select tests ... FAILED in 1 minutes and 3 seconds
{
  "name": "s3select:test_sql_operators",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "total_tests": 20,
    "total_success": 15
  },
  "duration": 60565,
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-12b5eb8d-24c2-4070-8554-cb82eae1b027, request_id: 1673A477499F6874, host_id: 2d7c3359-ceda-47a8-9258-9dfaaf77f9da, bucket_name: s3select-test-12b5eb8d-24c2-4070-8554-cb82eae1b027",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 61, in main\n    test_sql_operators(client, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 140, in test_sql_operators\n    raise select_err\n  File \"/mint/run/core/s3select/sql_ops.py\", line 138, in test_sql_operators\n    test_sql_expressions(client, json_testfile, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 91, in test_sql_expressions\n    input_serialization, output_serialization, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 78, in test_sql_expressions_custom_input_output\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-12b5eb8d-24c2-4070-8554-cb82eae1b027, request_id: 1673A477499F6874, host_id: 2d7c3359-ceda-47a8-9258-9dfaaf77f9da, bucket_name: s3select-test-12b5eb8d-24c2-4070-8554-cb82eae1b027\n",
  "status": "FAIL"
}
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="b6d4e594-8c37-41ca-a4c1-724e169ef4e9"></a>
### 12009-8b2eb4e/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:31905
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f6fa0b78551d:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 40 seconds
(4/15) Running aws-sdk-ruby tests ... done in 1 seconds
(5/15) Running awscli tests ... done in 2 minutes and 8 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 51 seconds
(8/15) Running minio-dotnet tests ... done in 32 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 16 seconds
(10/15) Running minio-java tests ... done in 26 seconds
(11/15) Running minio-js tests ... done in 43 seconds
(12/15) Running minio-py tests ... done in 1 minutes and 12 seconds
(13/15) Running s3cmd tests ... done in 12 seconds
(14/15) Running s3select tests ... FAILED in 5 minutes and 1 seconds
{
  "name": "s3select:test_sql_operators",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "total_tests": 20,
    "total_success": 15
  },
  "duration": 300312,
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-1a304f2a-327d-4f37-a6a8-6d1b5361df08, request_id: 1673A48561E0C8F3, host_id: f426e24b-94c3-4e80-88b1-c220549f6316, bucket_name: s3select-test-1a304f2a-327d-4f37-a6a8-6d1b5361df08",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 61, in main\n    test_sql_operators(client, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 140, in test_sql_operators\n    raise select_err\n  File \"/mint/run/core/s3select/sql_ops.py\", line 138, in test_sql_operators\n    test_sql_expressions(client, json_testfile, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 91, in test_sql_expressions\n    input_serialization, output_serialization, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 78, in test_sql_expressions_custom_input_output\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-1a304f2a-327d-4f37-a6a8-6d1b5361df08, request_id: 1673A48561E0C8F3, host_id: f426e24b-94c3-4e80-88b1-c220549f6316, bucket_name: s3select-test-1a304f2a-327d-4f37-a6a8-6d1b5361df08\n",
  "status": "FAIL"
}
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="9dd71fa2-d02e-42cf-a0b2-e6549b45579b"></a>
### 12009-8b2eb4e/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:31098
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5152404a07a3:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 2 minutes and 4 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 54 seconds
(8/15) Running minio-dotnet tests ... done in 32 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 26 seconds
(10/15) Running minio-java tests ... done in 40 seconds
(11/15) Running minio-js tests ... done in 54 seconds
(12/15) Running minio-py tests ... done in 1 minutes and 47 seconds
(13/15) Running s3cmd tests ... done in 17 seconds
(14/15) Running s3select tests ... FAILED in 1 minutes and 2 seconds
{
  "name": "s3select:test_sql_operators",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "total_tests": 20,
    "total_success": 15
  },
  "duration": 60292,
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-2ec3c7e5-1214-458e-83d0-411272f8cf10, request_id: 1673A45F4D7B091A, host_id: fe89ebe3-9cd4-43ea-a702-96f9bb79467a, bucket_name: s3select-test-2ec3c7e5-1214-458e-83d0-411272f8cf10",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 61, in main\n    test_sql_operators(client, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 140, in test_sql_operators\n    raise select_err\n  File \"/mint/run/core/s3select/sql_ops.py\", line 138, in test_sql_operators\n    test_sql_expressions(client, json_testfile, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 91, in test_sql_expressions\n    input_serialization, output_serialization, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 78, in test_sql_expressions_custom_input_output\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /s3select-test-2ec3c7e5-1214-458e-83d0-411272f8cf10, request_id: 1673A45F4D7B091A, host_id: fe89ebe3-9cd4-43ea-a702-96f9bb79467a, bucket_name: s3select-test-2ec3c7e5-1214-458e-83d0-411272f8cf10\n",
  "status": "FAIL"
}
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/bucket-lifecycle.go
+++ b/cmd/bucket-lifecycle.go
@@ -530,7 +530,7 @@ type SelectParameters struct {
 
 // IsEmpty returns true if no select parameters set
 func (sp *SelectParameters) IsEmpty() bool {
-	return sp == nil || sp.S3Select == s3select.S3Select{}
+	return sp == nil
 }
 
 var (

--- a/cmd/object-api-utils.go
+++ b/cmd/object-api-utils.go
@@ -817,13 +817,7 @@ func (g *GetObjectReader) Close() error {
 
 // Read - to implement Reader interface.
 func (g *GetObjectReader) Read(p []byte) (n int, err error) {
-	n, err = g.pReader.Read(p)
-	if err != nil {
-		// Calling code may not Close() in case of error, so
-		// we ensure it.
-		g.Close()
-	}
-	return
+	return g.pReader.Read(p)
 }
 
 //SealMD5CurrFn seals md5sum with object encryption key and returns sealed

--- a/pkg/s3select/select.go
+++ b/pkg/s3select/select.go
@@ -1,5 +1,5 @@
 /*
- * MinIO Cloud Storage, (C) 2019 MinIO, Inc.
+ * MinIO Cloud Storage, (C) 2019-2021 MinIO, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -216,6 +216,7 @@ type S3Select struct {
 	statement      *sql.SelectStatement
 	progressReader *progressReader
 	recordReader   recordReader
+	close          func() error
 }
 
 var (
@@ -311,6 +312,7 @@ func (s3Select *S3Select) Open(getReader func(offset, length int64) (io.ReadClos
 			}
 			return err
 		}
+		s3Select.close = rc.Close
 		return nil
 	case jsonFormat:
 		rc, err := getReader(0, -1)
@@ -333,6 +335,8 @@ func (s3Select *S3Select) Open(getReader func(offset, length int64) (io.ReadClos
 		} else {
 			s3Select.recordReader = json.NewReader(s3Select.progressReader, &s3Select.Input.JSONArgs)
 		}
+
+		s3Select.close = rc.Close
 		return nil
 	case parquetFormat:
 		if !strings.EqualFold(os.Getenv("MINIO_API_SELECT_PARQUET"), "on") {
@@ -396,6 +400,12 @@ func (s3Select *S3Select) marshal(buf *bytes.Buffer, record sql.Record) error {
 
 // Evaluate - filters and sends records read from opened reader as per select statement to http response writer.
 func (s3Select *S3Select) Evaluate(w http.ResponseWriter) {
+	defer func() {
+		if s3Select.close != nil {
+			s3Select.close()
+		}
+	}()
+
 	getProgressFunc := s3Select.getProgress
 	if !s3Select.Progress.Enabled {
 		getProgressFunc = nil

